### ISSUE TITLE
Optimizer can not work properly when using relative high temperature in inver problem

### ISSUE THREAD

**Issue by lss12343** [kailaix/ADCME.jl#78]
thanks for the ACDME here ,I just done with some inverse heattransfer problem ,but when i reach a pretty high temperature (above 1e2 k), the optimizer stop in a high loss value and the θ(thermal conductivity ) stop at a constant when temperature changing ( when i using a linear θ to get the data)

**Comment by lss12343** [comment]
I used both BFGS! and custom optimizer NLOPT! , and the same issue occurred ,especially i set the stopval in the NLOPT! , and it did not work .  here is the custom optimizer i used , which is very same in the toturial:

function NLOPT!(sess::PyObject, loss::PyObject, max_iter::Int64=15000; 
            algorithm::Union{Symbol, Enum} = :LD_LBFGS, vars::Array{PyObject}=PyObject[], 
                    callback::Union{Function, Nothing}=nothing, kwargs...)
    losses = Float64[]
    iter_ = 0 
    NLOPT = CustomOptimizer() do f, df, c, dc, x0, x_L, x_U 
        opt = Opt(algorithm, length(x0))
        opt.upper_bounds = x_U
        opt.lower_bounds = x_L
        opt.maxeval = max_iter
        opt.stopval = 1e-6
        opt.min_objective = (x,g)->begin
            g[:]= df(x); 
            iter_ += 1
            l = f(x)[1]
            println("================ ITER $iter_ ===============")
            println("current loss= $l")
            push!(losses, l)
            if !isnothing(callback)
                callback(run(sess, vars), iter_, l)
            end
            return f(x)[1]
        end
        (minf,minx,ret) = NLopt.optimize(opt, x0)
        minx
    end
    opt = NLOPT(loss; kwargs...)
    minimize(opt, sess)
    return losses
end
NLOPT!(sess, loss, algorithm = :LD_TNEWTON)
